presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkQueryPlanner.java

```diff
@@ -144,6 +147,11 @@ public PlanAndMore createQueryPlan(Session session, BuiltInPreparedQuery prepare
         List<String> columnNames = ((OutputNode) plan.getRoot()).getColumnNames();
         PhysicalResourceSettings physicalResourceSettings = new PrestoSparkPhysicalResourceCalculator()
                 .calculate(plan.getRoot(), new PrestoSparkSourceStatsCollector(metadata, session), session);
+        Map<FunctionKind, Set<String>> functionsInvoked = analysis.getInvokedFunctions();
+        Set<String> invokedScalarFunctions = functionsInvoked.getOrDefault(FunctionKind.SCALAR, new HashSet<>());
+        Set<String> invokedAggregateFunctions = functionsInvoked.getOrDefault(FunctionKind.AGGREGATE, new HashSet<>());
+        Set<String> invokedWindowFunctions = functionsInvoked.getOrDefault(FunctionKind.WINDOW, new HashSet<>());
```

Comment:
Is there a need to create the empty `HashSet` instance? Maybe `Collections.emptySet()` would be sufficient.